xcube_sh/sentinelhub.py

```diff
@@ -39,6 +38,55 @@
 from .version import version
 
 
+class SentinelOAuth2Session(requests_oauthlib.OAuth2Session):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def __getstate__(self):
+        return {
+            'client_id': self.client_id,
+            'compliance_hook': self.compliance_hook,
+            '_client': self._client,
+            'auto_refresh_url': self.auto_refresh_url,
+            'auto_refresh_kwargs': self.auto_refresh_kwargs,
+            'scope': self.scope,
+            'redirect_uri': self.redirect_uri,
+            'cookies': self.cookies,
+            'trust_env': self.trust_env,
+            'auth': self.auth,
+            'headers': self.headers,
+            'params': self.params,
+            'hooks': self.hooks,
+            'proxies': self.proxies,
+            'stream': self.stream,
+            'cert': self.cert,
+            'verify': self.verify,
+            'max_redirects': self.max_redirects,
+            'adapters': self.adapters
+        }
+
+    def __setstate__(self, state):
+        self._client = state['_client']
+        self.compliance_hook = state['compliance_hook']
+        self.client_id = state['client_id']
+        self.auto_refresh_url = state['auto_refresh_url']
+        self.auto_refresh_kwargs = state['auto_refresh_kwargs']
+        self.scope = state['scope']
+        self.trust_env = state['trust_env']
+        self.cookies = state['cookies']
+        self.redirect_uri = state['redirect_uri']
+        self.auth = state['auth']
+        self.headers = state['headers']
+        self.params = state['params']
+        self.hooks = state['hooks']
+        self.proxies = state['proxies']
+        self.stream = state['stream']
+        self.verify = state['verify']
+        self.cert = state['cert']
+        self.max_redirects = state['max_redirects']
+        self.adapters = state['adapters']
```

Comment:
Instead do:

    for a in self._SERIALIZED_ATTRS:
        setattr(self, a, state[a])